tree/splitter.h

```diff
@@ -0,0 +1,43 @@
+#ifndef _TREE_SPLITTER_H__
+#define _TREE_SPLITTER_H__
+
+/**
+ * Information regarding the split of a given node.
+ * This will be used by the tree during its construction.
+ */
+struct SplitterRecord
```

Comment:
/What will the method split return? An Interface of SplitterRecord ? It makes more sense, don't you think??